### DISCUSSION TITLE
How to enable graphics?

### DISCUSSION THREAD

**Discussion by edisonchan** [ProjectPhysX/FluidX3D#66] [Q&A] [answered]
I have try to uncomment INTERACTIVE_GRAPHICS in defnies.hpp, but still get a text mode benchmark .exe file only.
![image](https://user-images.githubusercontent.com/8596506/227690158-52ec0893-49ad-48a8-87d3-d9fde22b5925.png)

**Comment by ProjectPhysX** [comment, marked as answer]
In `src,defines.hpp`, comment `//#define BENCHMARK` and uncomment `#define INTERACTIVE_GRAPHICS`. Then, in `src/setup.cpp`, uncomment one of the `main_setup()` functions, compile and run. Depending on the setup, you might need to enable some extensions in `src/defines.hpp`: for example, for the "hydraulic jump" setup (text-search this in `src/setup.cpp`), you need to uncomment the extensions `VOLUME_FORCE`, `EQUILIBRIUM_BOUNDARIES` and `SURFACE`. The software will tell you in the console during startup if additional extensions are needed.

Once the graphics window opens, press P to start/pause the simulation. There is a help menu with key H. Have fun!
